Feature/disable email notifications

Eshwar1212-maker opened the pull request:
# What kind of change does this PR introduce?

Feature — Adds email notification toggling for users in the Settings UI to opt out of emails. This connects the existing backend logic (sendEmailToOrg) to the frontend with a working Button component.

eg: Bug fix, feature, docs update, ...

# Why was this change needed?

This is an active issue requested by a user, I will link it here:
https://github.com/gitroomhq/postiz-app/issues/163

# Other information:

I wanted to open a new request in the 'community support tickets' page on discord but currently don't have any permission.

# Checklist:

Put a "X" in the boxes below to indicate you have followed the checklist;

- [X] I have read the [CONTRIBUTING](https://github.com/gitroomhq/postiz-app/blob/main/CONTRIBUTING.md) guide.
- [X] I checked that there were not similar issues or PRs already open for this.
- [X

https://github.com/user-attachments/assets/c2e7043b-84ae-4005-a0cc-b7b65f3e65bf

<img width="1914" height="963" alt="Screenshot 2025-07-29 at 6 44 05 PM" src="https://github.com/user-attachments/assets/985f3f08-035a-4b7f-b265-f896951910aa" />
] This PR fixes just ONE issue (do not include multiple issues or types of change in the same PR) For example, don't try and fix a UI issue and include new dependencies in the same PR.

Mokkapps commented:
Thanks for the PR, I would love to have an option to receive email notifications for failed posts but not for successful posts.

nevo-david commented:
@Eshwar1212-maker, thank you!
But can you change it to the type of notifications, for example:
Failed post and succesful posts

Eshwar1212-maker commented:
Hey! Thanks for the feedback 

I’ve updated the UI section title to “Notifications” and kept the current toggle for email notifications under that section, as discussed. I just pushed this part . 

Just to clarify — the original ticket mentioned using a single boolean (`emailNotifications`) with a default of true, and I’ve followed that exactly. Do you want me to create an option for failed post/successful posts should I create an enum or JSON containing those values instead of having emailNotificaitons be a boolean?

And about failed/successful post preferences, are you saying that I should get rid of everything I implemented and redo it so users can just have options for failed/successful post preferences? I wanna clarify this because it goes against what's in the original ticket. And then also on top of that you want me to add a toggle for general notifications?

My current implementation followed the ticket word for word just for email notifications. I just wanna clarify if what I did is fine, but if you want me to add all the different features, you're asking it's almost a whole new ticket isn't it?

Thank you.
